### ISSUE TITLE
[Feature] [Serve] Accept kwargs in .deploy()

### ISSUE THREAD

**Issue by amiasato** [ray-project/ray#19022]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

This is not a blocking issue or anything, but it's a bit weird that the `deploy` method does not accept kwargs:

```python
from ray import serve

serve.start()

@serve.deployment(route_prefix="/")
class MyDeployment:
    def __init__(self, msg="foo"):
        print(msg)

MyDeployment.deploy()
MyDeployment.deploy(msg="bar")
```

```
(MyDeployment pid=39147) foo
(...)
TypeError: deploy() got an unexpected keyword argument 'msg'
```

Code works just fine if the keyword is passed as a regular argument. Switching the last line for:

```python
MyDeployment.deploy("bar")
```

```
(MyDeployment pid=39550) foo
(MyDeployment pid=39552) bar
```

Works just fine. I'm using the nightly wheel (2.0.0.dev0).

If this limitation is intended or simply not worth the hassle due to any implementation constraints, It would be nice to have it explicitly stated in the docs.

Thanks!

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by edoakes** [comment]
@amiasato I totally agree that this is "a bit weird" :) I think it was originally not supported due to the old API, but it should be pretty straightforward to add this now. Let me prioritize adding this.
